api/v1/installation_types.go

```diff
@@ -145,6 +148,14 @@ type InstallationSpec struct {
 	// NonPrivileged configures Calico to be run in non-privileged containers as non-root users where possible.
 	// +optional
 	NonPrivileged *NonPrivilegedType `json:"nonPrivileged,omitempty"`
+
+	// CalicoNodeDaemonset configures the calico-node DaemonSet. If used in
+	// conjunction with the deprecated ComponentResources, then these overrides take precedence.
+	// WARNING: Please note that this configuration overrides the operator's
+	// default calico-node DaemonSet configuration. The operator cannot validate
```

Comment:
I agree with you that there are a multitude of ways to break an operator-managed cluster via the API.

That said, I added this warning because I recalled that when we all chatted about the design that @tmjd wanted some warning in the API docs (I may be wrong). A plus is that this warning _might_ nudge users to think twice about their config changes.

Reply:
I'm not sure that we need a warning at this level. There is a lot that is configurable that we expect is normal and fairly safe to change.
I think we should probably be more targeted with our warnings, like I feel like we should have warnings on NodeSelector, Tolerations, and affinity.

Reply:
Addressed in 7e095012ebfd566145685e3fb5f32c619efaeb2b